tests/test_fixtures/rule_based_profiler/example_notebooks/BasicExample_RBP_Instantiation_and_running.ipynb

```diff
@@ -0,0 +1,1790 @@
+     "output_type": "stream",
+     "text": [
+      "Could not convert existing variables to dict: 'NoneType' object is not subscriptable\n"
+     ]
+    },
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "00d843b9a2e845c9a358d7a78047091b",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "Calculating Metrics:   0%|          | 0/2 [00:00<?, ?it/s]"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "res: ExpectationSuite = my_rbp.run()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "assert len(res.expectations) == 4"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "[{\"expectation_type\": \"expect_column_values_to_not_be_null\", \"meta\": {}, \"kwargs\": {\"column\": \"fare_amount\"}},\n",
+       " {\"expectation_type\": \"expect_column_values_to_not_be_null\", \"meta\": {}, \"kwargs\": {\"column\": \"tip_amount\"}},\n",
+       " {\"expectation_type\": \"expect_column_values_to_not_be_null\", \"meta\": {}, \"kwargs\": {\"column\": \"tolls_amount\"}},\n",
+       " {\"expectation_type\": \"expect_column_values_to_not_be_null\", \"meta\": {}, \"kwargs\": {\"column\": \"total_amount\"}}]"
+      ]
+     },
+     "execution_count": 14,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "res.expectations"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As expected our simple RuleBasedProfiler will output an `ExpectationSuite` with 4 Expectations, one for each of our 4 columns. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Example 2: RBP with `DomainBuilder`, `ParameterBuilder` `ExpectationConfigurationBuilder`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Build a DomainBuilder\n",
+    "* Using same `SimpleColumnSuffixDomainBuilder` from previous example."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 15,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "77fb10cfb7644f459a7a25a73238acb5",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "Calculating Metrics:   0%|          | 0/2 [00:00<?, ?it/s]"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "domain_builder: DomainBuilder = SimpleColumnSuffixDomainBuilder(\n",
+    "    data_context=data_context,\n",
+    "    batch_request=single_batch_batch_request,\n",
+    "    column_name_suffixes=[\"_amount\"],\n",
+    ")\n",
+    "domains: list = domain_builder.get_domains()"
```

Comment:
When I ran this cell in the notebook it errored out, with the following traceback:

```
---------------------------------------------------------------------------
DatasourceError                           Traceback (most recent call last)
Input In [13], in <cell line: 6>()
      1 domain_builder: DomainBuilder = SimpleColumnSuffixDomainBuilder(
      2     data_context=data_context,
      3     batch_request=single_batch_batch_request,
      4     column_name_suffixes=["_amount"],
      5 )
----> 6 domains: list = domain_builder.get_domains()

File ~/PycharmProjects/great_expectations/great_expectations/rule_based_profiler/domain_builder/domain_builder.py:54, in DomainBuilder.get_domains(self, variables)
     46 def get_domains(
     47     self,
     48     variables: Optional[ParameterContainer] = None,
     49 ) -> List[Domain]:
     50     """
     51     Note: Please do not overwrite the public "get_domains()" method.  If a child class needs to check parameters,
     52     then please do so in its implementation of the (private) "_get_domains()" method, or in a utility method.
     53     """
---> 54     return self._get_domains(variables=variables)

File ~/PycharmProjects/great_expectations/great_expectations/rule_based_profiler/domain_builder/simple_column_suffix_domain_builder.py:62, in SimpleColumnSuffixDomainBuilder._get_domains(self, variables)
     55 def _get_domains(
     56     self,
     57     variables: Optional[ParameterContainer] = None,
     58 ) -> List[Domain]:
     59     """
     60     Find the column suffix for each column and return all domains matching the specified suffix.
     61     """
---> 62     table_column_names: List[str] = self.get_effective_column_names(
     63         include_columns=self.column_names,
     64         exclude_columns=None,
     65         variables=variables,
     66     )
     68     # Obtain column_name_suffixes from "rule state" (i.e., variables and parameters); from instance variable otherwise.
     69     column_name_suffixes: Union[
     70         str, Iterable, List[str]
     71     ] = get_parameter_value_and_validate_return_type(
   (...)
     76         parameters=None,
     77     )

File ~/PycharmProjects/great_expectations/great_expectations/rule_based_profiler/domain_builder/column_domain_builder.py:79, in ColumnDomainBuilder.get_effective_column_names(self, include_columns, exclude_columns, variables)
     70 # Obtain exclude_columns from "rule state" (i.e., variables and parameters); from instance variable otherwise.
     71 exclude_columns = get_parameter_value_and_validate_return_type(
     72     domain=None,
     73     parameter_reference=exclude_columns,
   (...)
     76     parameters=None,
     77 )
---> 79 batch_ids: List[str] = self.get_batch_ids(variables=variables)
     81 table_columns: List[str] = self.get_validator(variables=variables).get_metric(
     82     metric=MetricConfiguration(
     83         metric_name="table.columns",
   (...)
     89     )
     90 )
     92 effective_column_names: List[str] = include_columns or table_columns

File ~/PycharmProjects/great_expectations/great_expectations/rule_based_profiler/domain_builder/domain_builder.py:141, in DomainBuilder.get_batch_ids(self, variables)
    137 def get_batch_ids(
    138     self,
    139     variables: Optional[ParameterContainer] = None,
    140 ) -> Optional[List[str]]:
--> 141     return get_batch_ids_from_batch_list_or_batch_request(
    142         data_context=self.data_context,
    143         batch_list=self.batch_list,
    144         batch_request=self.batch_request,
    145         domain=None,
    146         variables=variables,
    147         parameters=None,
    148     )

File ~/PycharmProjects/great_expectations/great_expectations/rule_based_profiler/helpers/util.py:110, in get_batch_ids(data_context, batch_list, batch_request, domain, variables, parameters)
    101         return None
    103     batch_request = build_batch_request(
    104         domain=domain,
    105         batch_request=batch_request,
    106         variables=variables,
    107         parameters=parameters,
    108     )
--> 110     batch_list = data_context.get_batch_list(batch_request=batch_request)
    112 batch_ids: List[str] = [batch.id for batch in batch_list]
    114 num_batch_ids: int = len(batch_ids)

File ~/PycharmProjects/great_expectations/great_expectations/core/usage_statistics/usage_statistics.py:327, in usage_statistics_enabled_method.<locals>.usage_statistics_wrapped_method(*args, **kwargs)
    324     if args_payload_fn is not None:
    325         nested_update(event_payload, args_payload_fn(*args, **kwargs))
--> 327     result = func(*args, **kwargs)
    328     message["success"] = True
    329 except Exception:

File ~/PycharmProjects/great_expectations/great_expectations/data_context/data_context.py:1760, in BaseDataContext.get_batch_list(self, datasource_name, data_connector_name, data_asset_name, batch_request, batch_data, data_connector_query, batch_identifiers, limit, index, custom_filter_function, sampling_method, sampling_kwargs, splitter_method, splitter_kwargs, runtime_parameters, query, path, batch_filter_parameters, batch_spec_passthrough, **kwargs)
   1758     datasource: Datasource = cast(Datasource, self.datasources[datasource_name])
   1759 else:
-> 1760     raise ge_exceptions.DatasourceError(
   1761         datasource_name,
   1762         "The given datasource could not be retrieved from the DataContext; please confirm that your configuration is accurate.",
   1763     )
   1764 return datasource.get_batch_list_from_batch_request(batch_request=batch_request)

DatasourceError: Cannot initialize datasource taxi_multibatch_datasource_other_possibility, error: The given datasource could not be retrieved from the DataContext; please confirm that your configuration is accurate.
```

Since I hadn't made any changes to the config, this seems like something that needs to be reviewed if this notebook is intended to run out of the box.

Reply:
So sorry about this. This was a misname of the `datasource`. Has been resolved in most recent commit